pkg/vault/contracts/VaultCommon.sol

```diff
@@ -300,22 +309,16 @@ abstract contract VaultCommon is IVaultEvents, IVaultErrors, VaultStorage, Reent
 
     /**
      * @dev Fill in PoolData, including paying protocol yield fees and computing final raw and live balances.
-     * This function modifies protocol fees and balance storage. Since it modifies storage and makes external
-     * calls, it must be nonReentrant.
+     * This function modifies protocol fees and balance storage. Out of an abundance of caution, since `_loadPoolData`
+     * makes external calls, we are making anything that calls it and then modifies storage non-reentrant.
      * Side effects: updates `_aggregateFeeAmounts` and `_poolTokenBalances` in storage.
      */
     function _loadPoolDataUpdatingBalancesAndYieldFees(
         address pool,
         Rounding roundingDirection
     ) internal nonReentrant returns (PoolData memory poolData) {
         // Initialize poolData with base information for subsequent calculations.
-        poolData.load(
-            _poolTokenBalances[pool],
-            _poolConfigBits[pool],
-            _poolTokenInfo[pool],
-            _poolTokens[pool],
-            roundingDirection
-        );
+        poolData = _loadPoolData(pool, roundingDirection);
```

Comment:
Looks like this change increased the vault bytecode by 23 bytes. Is it really needed?

Reply:
I'll check if that's really what did it. Seems counter-intuitive that calling a function vs. duplicating that function inline *increases* the bytecode - it was intended to 1) remove duplication (and ensure what should be identical code doesn't diverge); and 2) decrease bytecode by sharing code. But the optimizer is voodoo, so who knows. If I can save 23 bytes, I'll put it back.